libsawtooth/src/state/merkle.rs

```diff
@@ -38,6 +42,144 @@ pub fn decode_cbor_value(cbor_bytes: &[u8]) -> Result<Vec<u8>, StateDatabaseErro
     }
 }
 
+pub fn encode_cbor_value(bytes: &[u8]) -> Result<Vec<u8>, StateDatabaseError> {
+    let mut encoder = GenericEncoder::new(Cursor::new(Vec::new()));
+    encoder
+        .value(&Value::Bytes(Bytes::Bytes(bytes.to_vec())))
+        .map_err(|_| StateDatabaseError::InvalidRecord)?;
+    Ok(encoder.into_inner().into_writer().into_inner())
+}
+
+#[derive(Clone)]
+pub struct CborMerkleState {
+    db: Box<dyn Database>,
```

Comment:
Could this just wrap a `MerkleRadixTree`?

Reply:
No, I tried that originally but commit and update need to be mut, but the trait for Write/Read takes &self.

Reply:
Sorry, looking at the source code, I think I meant `MerkleState`